tests/query/processors/test_mandatory_condition_applier.py

```diff
@@ -0,0 +1,80 @@
+import pytest
+from typing import List
+
+# from snuba.clickhouse.columns import ColumnSet, String
+
+from snuba.datasets.schemas.tables import TableSource
+from snuba.query.logical import Query
+from snuba.query.processors.mandatory_condition_applier import MandatoryConditionApplier
+
+from snuba.request.request_settings import HTTPRequestSettings
+from snuba.datasets.schemas import MandatoryCondition
+from snuba.query.conditions import ConditionFunctions, binary_condition
+from snuba.query.expressions import Column, Literal
+
+
+test_data = [
+    (
+        "table1",
+        [
+            MandatoryCondition(
+                ["deleted", "=", "0"],
+                binary_condition(
+                    None,
+                    ConditionFunctions.EQ,
+                    Column(None, None, "deleted"),
+                    Literal(None, 0),
+                ),
+            )
+        ],
+    ),
+    (
+        "table2",
+        [
+            MandatoryCondition(
+                ["time", "=", "1"],
+                binary_condition(
+                    None,
+                    ConditionFunctions.EQ,
+                    Column(None, None, "time"),
+                    Literal(None, 0),
+                ),
+            ),
+            MandatoryCondition(
+                ["time2", "=", "2"],
+                binary_condition(
+                    None,
+                    ConditionFunctions.EQ,
+                    Column(None, None, "time2"),
+                    Literal(None, 0),
+                ),
+            ),
+        ],
+    ),
+]
+
+
+@pytest.mark.parametrize("table, mand_condition", test_data)
+def test_mand_condition(table: str, mand_condition: List[MandatoryCondition]) -> None:
+
+    body = {
+        "conditions": [
+            ["d", "=", "1"],
+            ["c", "=", "3"],
+            ["a", "=", "1"],
+            ["b", "=", "2"],
+        ],
+    }
+
+    # cols = ColumnSet([("col", String())])
+
+    cols = None
+    consistent = True
+
+    query = Query(body, TableSource(table, cols, mand_condition, ["c1"]),)
+
+    request_settings = HTTPRequestSettings(consistent=consistent)
+    processor = MandatoryConditionApplier()
+    processor.process_query(query, request_settings)
+
+    assert query.get_conditions() == body["conditions"]
```

Comment:
While this passes, this condition does not test exactly what you want:
The body of the query is a dictionary (thus mutable), then you create the query with that dictionary, which means the query holds a reference to the dictionary you built, then the query processor adds some entries to that dictionary. The dictionary is always the same, none made copies.

Now when you assert `query.get_conditions() == body["conditions"]` you are actually comparing an object with itself (which will always pass no matter what the query processor does.

The way to test this condition is, on line 71 to pass a copy of the dictionary (with the deepcopy function) to the Query initializer instead of the same dictionary.
Then you will be comparing what you want, and the test will fail because `get_conditions` will return some different content since it has been transformed.
This is because your test is missing one element, the expected result you should compare the result of your processor with.

When you build a unit test, you generally approach it this way, let's say you are trying to test a function that multiples a number by itself:
```
def test_something() -> None:
   input = 5 
  expected_output = 25

  assert function_to_test(input) == expected_output
```

When you parametrize your test by defining multiple cases, for each case in test_data you should add an attribute with the expected result, then pass the input to your processor and assert the output is equal to the expected result.